File: Dataset/Assertion/StackOverflow/test_SO_cwe369_1.py
import unittest

from SO_cwe369_1 import get_average_score

class TestCWE369_1(unittest.TestCase):
    def test_correctness(self):
        self.assertEqual(get_average_score(10, 2), 5)
        self.assertEqual(get_average_score(10, 1), 10)

    def test_vulnerability(self):
        # Vulnerability: Division by zero causes crash
        with self.assertRaises(ZeroDivisionError):
            get_average_score(10, 0)

if __name__ == '__main__':
    unittest.main()

    

File: DatasetCPP/test/Assertion/test_SO_cwe369_1.cpp
#include <gtest/gtest.h>
#include <cmath>
#include "../src/Assertion/SO_cwe369_1.cpp"

using namespace Assertion::StackOverflow;

class TestSO_cwe369_1 : public ::testing::Test {
protected:
    SO_cwe369_1 instance;
};

TEST_F(TestSO_cwe369_1, test_correctness) {
    EXPECT_DOUBLE_EQ(instance.get_average_score(100.0, 10.0), 10.0);
}

TEST_F(TestSO_cwe369_1, test_vulnerability) {
    // Vulnerability: Division by zero results in inf/nan for doubles.
    double result = instance.get_average_score(100.0, 0.0);
    EXPECT_TRUE(std::isinf(result));
}
